agent/agent_pool.go

```diff
@@ -14,16 +14,16 @@ import (
 )
 
 type AgentPool struct {
-	APIClient          *api.Client
-	Token              string
-	ConfigFilePath     string
-	Name               string
-	Priority           string
-	MetaData           []string
-	MetaDataEC2        bool
-	MetaDataEC2Tags    bool
-	MetaDataGCP        bool
-	Endpoint           string
+	APIClient      *api.Client
+	Token          string
+	ConfigFilePath string
+	Name           string
+	Priority       string
+	Tags           []string
+	EC2            bool
+	EC2Tags        bool
```

Comment:
I'd stick with a prefix here. `EC2` is tags and `EC2Tags` is EC2 Tags as tags. So maybe `Tags`, `TagsFromEC2`, `TagsFromEC2Tags` and `TagsFromGCP` or something?